### PR TITLE
workload: Support multiple cloud storage providers.

### PR DESCRIPTION
**Pull request by miretskiy**
Loading previously created fixtures speeds up data loading stage.
However, previuously, `workload fixture (make|load)` only supported GS
storage.  This change enhances workload to support s3:// and azure:// storage.

To keep backward compatibility, the default remaining GS.  However, it is now
possible to configure workload to read/write other storage providers:

```
workload fixtures  make  tpcc --warehouses=10000 --provider-override=s3 --auth-params-override="AUTH=implicit&AWS_REGION=us-east-1"
```

The `--provider-override` flag overrides the storage provider, which by default,
remaians "gs://".  Other valid providers include `s3` and `azure`.

The `--auth-params-override` flag overrides default authentication parameters.
These are cloud provider specific.  For example, AWS requires that the region
is specified.  See cloud storage documentation for more details on the supported
storage provider configurations.

Release Notes: None

### CODE REVIEW

**Review by ZhouXing19 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74883)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @miretskiy and @tbg)

___
*[pkg/ccl/workloadccl/fixture_test.go, line 124 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtQdbU342rGiPs1CuTD:-MtQd_SK-qd6KXuImnNl:b-u0fhsd) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture_test.go#L124)):*
> ```Go
> 	}
> 
> 	if _, err := workloadccl.GetFixture(ctx, gcs, config, gen); !testutils.IsError(err, `fixture not found`) {
> ```

Since the signature of `workloadccl.GetFixture` has been modified, we need to change the `gcs` (of `storage.NewClient` type) in the parameter list to a `cloud.ExternalStorage` type.  
___
*[pkg/ccl/workloadccl/fixture_test.go, line 128 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtQe8j6Bj7fGhBSvCDr:-MtQe8j6Bj7fGhBSvCDs:b-7ropln) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture_test.go#L128)):*
> ```Go
> 	}
> 
> 	fixtures, err := workloadccl.ListFixtures(ctx, gcs, config)
> ```

ditto
___
*[pkg/ccl/workloadccl/fixture_test.go, line 137 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtQe9f_C3NzX1K-4b_Z:-MtQe9f_C3NzX1K-4b__:b-7ropln) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture_test.go#L137)):*
> ```Go
> 
> 	const filesPerNode = 1
> 	fixture, err := workloadccl.MakeFixture(ctx, db, gcs, config, gen, filesPerNode)
> ```

ditto
___
*[pkg/ccl/workloadccl/fixture_test.go, line 142 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtQeAW45zq2zlrfD1GX:-MtQeAW45zq2zlrfD1GY:b-7ropln) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture_test.go#L142)):*
> ```Go
> 	}
> 
> 	_, err = workloadccl.MakeFixture(ctx, db, gcs, config, gen, filesPerNode)
> ```

ditto
___
*[pkg/ccl/workloadccl/fixture_test.go, line 147 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtQeBRS5MZGwKInsBCY:-MtQeBRS5MZGwKInsBCZ:b-7ropln) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture_test.go#L147)):*
> ```Go
> 	}
> 
> 	fixtures, err = workloadccl.ListFixtures(ctx, gcs, config)
> ```

ditto

**Comment by miretskiy** [comment]
You can probably tell I was testing things live.  Test updated.

**Review by ZhouXing19 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74883)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @miretskiy, @otan, and @tbg)

___
*[pkg/ccl/workloadccl/fixture.go, line 47 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtiNSD0EtGlW1aKggyp:-MtiNSD0EtGlW1aKggyq:b-x5uugb) ([raw file](https://github.com/cockroachdb/cockroach/blob/5094e02c6575b5c27b1b0547b9e80ecd54e605f3/pkg/ccl/workloadccl/fixture.go#L47)):*
> ```Go
> 	Bucket string
> 
> 	// Basename is appended to the bucket to form a path to the fixtures directory.
> ```

The `Basename` field is to rename the original `GCSPrefix`, right? If so, then it should still be "prepended" instead of "appended".
___
*[pkg/ccl/workloadccl/fixture.go, line 125 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtiO2PWC3IrAAeJdOp2:-MtiO2PWC3IrAAeJdOp3:bpo01my) ([raw file](https://github.com/cockroachdb/cockroach/blob/5094e02c6575b5c27b1b0547b9e80ecd54e605f3/pkg/ccl/workloadccl/fixture.go#L125)):*
> ```Go
> func generatorToStorageFolder(config FixtureConfig, gen workload.Generator) string {
> 	meta := gen.Meta()
> 	return filepath.Join(meta.Name,
> ```

Are we missing `config.Basename` here? If not, we can just remove `config FixtureConfig` in the parameter list.
___
*[pkg/ccl/workloadccl/fixture.go, line 129 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtiOI_i6TT_i_BEW5DO:-MtiOI_i6TT_i_BEW5DP:b-jgm7i1) ([raw file](https://github.com/cockroachdb/cockroach/blob/5094e02c6575b5c27b1b0547b9e80ecd54e605f3/pkg/ccl/workloadccl/fixture.go#L129)):*
> ```Go
> }
> 
> // FixtureURL returns the URL for pre-computed Generator data stored on GCS.
> ```

Now it's not dedicated for GCS. Maybe change it to `on given cloud service provider`.
___
*[pkg/ccl/workloadccl/fixture.go, line 134 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-Mtir_Iz8tkobrx2L_Ir:-Mtir_Iz8tkobrx2L_Is:b5y2ujg) ([raw file](https://github.com/cockroachdb/cockroach/blob/5094e02c6575b5c27b1b0547b9e80ecd54e605f3/pkg/ccl/workloadccl/fixture.go#L134)):*
> ```Go
> }
> 
> // GetFixture returns a handle for pre-computed Generator data stored on GCS. It
> ```

ditto, it's not GCS-dedicated now.
___
*[pkg/ccl/workloadccl/fixture.go, line 151 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtidveGEJ0C9NtJ7q-U:-MtidveGEJ0C9NtJ7q-V:b-i0a352) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture.go#L151)):*
> ```Go
> 
> 			fixtureFolder := generatorToGCSFolder(config, gen)
> 			_, err := b.Objects(ctx, &storage.Query{Prefix: fixtureFolder, Delimiter: `/`}).Next()
> ```

Just curious why we're removing the `retry` logic, and why we don't need to confirm that `fixtureFolder` exists now? 
___
*[pkg/ccl/workloadccl/fixture.go, line 653 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtisC_Y99wkPW2xNo_5:-MtisC_Y99wkPW2xNo_6:b-x6g6j6) ([raw file](https://github.com/cockroachdb/cockroach/blob/5094e02c6575b5c27b1b0547b9e80ecd54e605f3/pkg/ccl/workloadccl/fixture.go#L653)):*
> ```Go
> }
> 
> func listDir(
> ```

This function description is missed here.
And maybe rename `dirs` to `pathPrefixes`?  
___
*[pkg/ccl/workloadccl/fixture_test.go, line 93 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtizZtCC5TCLW8r2U1S:-MtizZtCC5TCLW8r2U1T:b-blln2v) ([raw file](https://github.com/cockroachdb/cockroach/blob/5094e02c6575b5c27b1b0547b9e80ecd54e605f3/pkg/ccl/workloadccl/fixture_test.go#L93)):*
> ```Go
> 
> 	if storageProvider == "" {
> 		skip.IgnoreLint(t, "FIXTURE_TEST_STORAGE_PROVIDER env var must be set")
> ```

Should be `COCKROACH_FIXTURE_TEST_STORAGE_PROVIDER`

**Review by miretskiy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74883)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @miretskiy, @otan, @tbg, and @ZhouXing19)

___
*[pkg/ccl/workloadccl/fixture.go, line 47 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtiNSD0EtGlW1aKggyp:-Mtj8TkJB98c2WHzbDmY:bhkjkr8) ([raw file](https://github.com/cockroachdb/cockroach/blob/5094e02c6575b5c27b1b0547b9e80ecd54e605f3/pkg/ccl/workloadccl/fixture.go#L47)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

The `Basename` field is to rename the original `GCSPrefix`, right? If so, then it should still be "prepended" instead of "appended".
</blockquote></details>

No; it is appended to the bucket name.  "gs://bucket-name/basename/..."
___
*[pkg/ccl/workloadccl/fixture.go, line 125 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtiO2PWC3IrAAeJdOp2:-Mtj8d2t2ChtxBNULN7K:bj9ukhc) ([raw file](https://github.com/cockroachdb/cockroach/blob/5094e02c6575b5c27b1b0547b9e80ecd54e605f3/pkg/ccl/workloadccl/fixture.go#L125)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

Are we missing `config.Basename` here? If not, we can just remove `config FixtureConfig` in the parameter list.
</blockquote></details>

No we are not.  Granted, it's a bit confusing as to where and how these bits are used;
This is (essentially) what `GetStorage` method does:
```
s, err := cloud.ExternalStorageFromURI(ctx, cfg.ObjectPathToURI(), ...
```

And the interesting bit is the `ObjectPathToURI`:
```
func (s FixtureConfig) ObjectPathToURI(folders ...string) string {
	path := filepath.Join(folders...)
	return fmt.Sprintf("%s://%s/%s/%s?%s", s.StorageProvider, s.Bucket, s.Basename, path, s.AuthParams)
}
```

Basically, the "cloud storage" will return an object that will assume that everything is below "bucket/basename".
So, we don't need to append basename here.
___
*[pkg/ccl/workloadccl/fixture.go, line 129 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtiOI_i6TT_i_BEW5DO:-Mtj9WwpBr9a3Kh9Itdq:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5094e02c6575b5c27b1b0547b9e80ecd54e605f3/pkg/ccl/workloadccl/fixture.go#L129)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

Now it's not dedicated for GCS. Maybe change it to `on given cloud service provider`.
</blockquote></details>

Done.
___
*[pkg/ccl/workloadccl/fixture.go, line 134 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-Mtir_Iz8tkobrx2L_Ir:-Mtj9__Z4548o2SJrZ0Z:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5094e02c6575b5c27b1b0547b9e80ecd54e605f3/pkg/ccl/workloadccl/fixture.go#L134)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

ditto, it's not GCS-dedicated now.
</blockquote></details>

Done.
___
*[pkg/ccl/workloadccl/fixture.go, line 151 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtidveGEJ0C9NtJ7q-U:-Mtj9aUq-nro16lC0qgn:b-ykcbjt) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture.go#L151)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

Just curious why we're removing the `retry` logic, and why we don't need to confirm that `fixtureFolder` exists now? 
</blockquote></details>

cloudstorage implementations have internal retries.
___
*[pkg/ccl/workloadccl/fixture_test.go, line 124 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtQdbU342rGiPs1CuTD:-Mtj9kzLCQZGYW7uL_m5:bt1b19p) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture_test.go#L124)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

Since the signature of `workloadccl.GetFixture` has been modified, we need to change the `gcs` (of `storage.NewClient` type) in the parameter list to a `cloud.ExternalStorage` type.  
</blockquote></details>

Renamed.
___
*[pkg/ccl/workloadccl/fixture_test.go, line 128 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtQe8j6Bj7fGhBSvCDr:-Mtj9nOo5My7uvEkX8Sf:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture_test.go#L128)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.
___
*[pkg/ccl/workloadccl/fixture_test.go, line 137 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtQe9f_C3NzX1K-4b_Z:-Mtj9nta0amTq3lKYWgu:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture_test.go#L137)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.
___
*[pkg/ccl/workloadccl/fixture_test.go, line 142 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtQeAW45zq2zlrfD1GX:-Mtj9oMz94QyTtV2EDJC:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture_test.go#L142)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.
___
*[pkg/ccl/workloadccl/fixture_test.go, line 147 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtQeBRS5MZGwKInsBCY:-Mtj9olIEJUkxeeggfvH:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture_test.go#L147)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.
___
*[pkg/ccl/workloadccl/fixture_test.go, line 93 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtizZtCC5TCLW8r2U1S:-Mtj9gei8z4XDuOnRo5q:b-wtkc5h) ([raw file](https://github.com/cockroachdb/cockroach/blob/5094e02c6575b5c27b1b0547b9e80ecd54e605f3/pkg/ccl/workloadccl/fixture_test.go#L93)):*
<details><summary><i>Previously, ZhouXing19 (Jane Xing) wrote…</i></summary><blockquote>

Should be `COCKROACH_FIXTURE_TEST_STORAGE_PROVIDER`
</blockquote></details>

Yes, thank you.

**Review by miretskiy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74883)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @miretskiy, @otan, @tbg, and @ZhouXing19)

___
*[pkg/ccl/workloadccl/fixture.go, line 125 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtiO2PWC3IrAAeJdOp2:-MtjBIJL1o5RNnNI9O3H:bcqud6y) ([raw file](https://github.com/cockroachdb/cockroach/blob/5094e02c6575b5c27b1b0547b9e80ecd54e605f3/pkg/ccl/workloadccl/fixture.go#L125)):*
<details><summary><i>Previously, miretskiy (Yevgeniy Miretskiy) wrote…</i></summary><blockquote>

No we are not.  Granted, it's a bit confusing as to where and how these bits are used;
This is (essentially) what `GetStorage` method does:
```
s, err := cloud.ExternalStorageFromURI(ctx, cfg.ObjectPathToURI(), ...
```

And the interesting bit is the `ObjectPathToURI`:
```
func (s FixtureConfig) ObjectPathToURI(folders ...string) string {
	path := filepath.Join(folders...)
	return fmt.Sprintf("%s://%s/%s/%s?%s", s.StorageProvider, s.Bucket, s.Basename, path, s.AuthParams)
}
```

Basically, the "cloud storage" will return an object that will assume that everything is below "bucket/basename".
So, we don't need to append basename here.


</blockquote></details>

Parameter removed.

**Comment by miretskiy** [comment]
> > func listDir(
> > ```
> 
> This function description is missed here. And maybe rename `dirs` to `pathPrefixes`?
> 

Kept as listDir -- realized that I no longer need variadic dirs -- so renamed just to "dir"

**Review by srosenberg — APPROVED** [review]
:lgtm:

> Reviewed 4 of 6 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-:-MtkVVDpE6SmsZg-RKoL:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @miretskiy, @otan, @tbg, and @ZhouXing19)

___
*[pkg/ccl/workloadccl/fixture.go, line 151 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtidveGEJ0C9NtJ7q-U:-MtkUnAFAkwrVNum9502:b5tinf9) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture.go#L151)):*
<details><summary><i>Previously, miretskiy (Yevgeniy Miretskiy) wrote…</i></summary><blockquote>

cloudstorage implementations have internal retries.
</blockquote></details>

@zhouxing19 As Yevgeniy said, retry is handled internally by corresponding cloudstorage impl. Existence is checked by noting `len(files)` after the corresponding `fixtureFolder` is enumerated for matching blobs.
___
*[pkg/ccl/workloadccl/cliccl/fixtures.go, line 39 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtkRFrd1essR7v4Q7Nh:-MtkRFrd1essR7v4Q7Ni:b7h1uws) ([raw file](https://github.com/cockroachdb/cockroach/blob/337b3b15412895a2219da436ad7212a32255925c/pkg/ccl/workloadccl/cliccl/fixtures.go#L39)):*
> ```Go
> 	config := defaultConfig
> 	if len(*providerOverride) > 0 {
> 		config.StorageProvider = *providerOverride
> ```

Maybe add to `ToLower` since provider name is typically case-insensitive.

**Review by ZhouXing19 — APPROVED** [review]
Thanks for the changes!
My only concern is for the test -- since we modified the required env vars in `TestFixture` at `pkg/ccl/workloadccl/fixture_test.go`, we may need to configure them in the CI, otherwise `TestFixture` will always be skipped:

```
storageProvider := envutil.EnvOrDefaultString("COCKROACH_FIXTURE_TEST_STORAGE_PROVIDER", "")
if storageProvider == "" {
		skip.IgnoreLint(t, "COCKROACH_FIXTURE_TEST_STORAGE_PROVIDER env var must be set")
}
```
Other than that, :lgtm: !

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-:-MtkX2lwAz6lKn3ai8NM:bcn02w2)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @miretskiy, @otan, @srosenberg, @tbg, and @ZhouXing19)

___
*[pkg/ccl/workloadccl/fixture.go, line 125 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtiO2PWC3IrAAeJdOp2:-MtkO7Y47AEnl1CaYNL-:b7six7g) ([raw file](https://github.com/cockroachdb/cockroach/blob/5094e02c6575b5c27b1b0547b9e80ecd54e605f3/pkg/ccl/workloadccl/fixture.go#L125)):*
<details><summary><i>Previously, miretskiy (Yevgeniy Miretskiy) wrote…</i></summary><blockquote>

Parameter removed.
</blockquote></details>

Ah I see, thanks!
___
*[pkg/ccl/workloadccl/fixture.go, line 151 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtidveGEJ0C9NtJ7q-U:-MtkWY121hnS--TDUq9B:b4m8pki) ([raw file](https://github.com/cockroachdb/cockroach/blob/ced473fa0ed00660812a6f6678155fb30e6a871f/pkg/ccl/workloadccl/fixture.go#L151)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

@zhouxing19 As Yevgeniy said, retry is handled internally by corresponding cloudstorage impl. Existence is checked by noting `len(files)` after the corresponding `fixtureFolder` is enumerated for matching blobs.
</blockquote></details>

TIL, thanks!
___
*[pkg/ccl/workloadccl/fixture_test.go, line 97 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/74883#-MtkWc924qovbTQQo3md:-MtkWc924qovbTQQo3me:bw03c06) ([raw file](https://github.com/cockroachdb/cockroach/blob/337b3b15412895a2219da436ad7212a32255925c/pkg/ccl/workloadccl/fixture_test.go#L97)):*
> ```Go
> 
> 	es, err := workloadccl.GetStorage(ctx, workloadccl.FixtureConfig{
> 		StorageProvider: "gs",
> ```

Should we use `StorageProvider: storageProvider,` here?

**Comment by ZhouXing19** [comment]
We ran with all three storages -- `gs`, `s3`, and `azure` and they all passed tests in `pkg/ccl/workloadccl`. 
Specifically: 
- gs: `COCKROACH_FIXTURE_TEST_STORAGE_PROVIDER=gs go test ./pkg/ccl/workloadccl`

- aws: `COCKROACH_FIXTURE_TEST_AUTH_PARAMS="AUTH=implicit&AWS_REGION=us-east-1" COCKROACH_FIXTURE_TEST_STORAGE_PROVIDER=s3 go test ./pkg/ccl/workloadccl`

- azure: `COCKROACH_FIXTURE_TEST_AUTH_PARAMS="AZURE_ACCOUNT_NAME=fixturetest&AZURE_ACCOUNT_KEY=the_secret_key" COCKROACH_FIXTURE_TEST_STORAGE_PROVIDER=azure go test ./pkg/ccl/workloadccl`. Replace the `the_secret_key` with key obtained from running `az storage account keys list --account-name=fixturetest` in the cli.

**Comment by ZhouXing19** [comment]
Note that for an `azure` use case:
- Go to the `Storage accounts` page of azure portal and create a Storage Account (e.g. MyAccount)
- Go to the account page of `MyAccount` and click on the `containers` on the left panel, create a container (e.g. MyContainer)
- Run `az storage account keys list --account-name=MyAccount` to get the account key
- Run `COCKROACH_FIXTURE_TEST_AUTH_PARAMS="AZURE_ACCOUNT_NAME=MyAccount&AZURE_ACCOUNT_KEY=the_secret_key"  COCKROACH_FIXTURE_TEST_BUCKET="MyContainer"  COCKROACH_FIXTURE_TEST_STORAGE_PROVIDER=azure go test ./pkg/ccl/workloadccl` for the test.
